ts/index.ts

```diff
@@ -0,0 +1,196 @@
+    this.visibility = this.determineVisibility();
+    this.tag = inputs.getString("tag");
+    this.host = inputs.getString("host");
+    this.logDirectives = inputs.getString("log-directives");
+    this.logger = inputs.getString("logger");
+    this.gitHubToken = inputs.getString("github-token");
+    this.repository = inputs.getString("repository");
+    this.directory = inputs.getString("directory");
+    this.gitRoot = inputs.getString("git-root");
+    // extra-tags is deprecated but we still honor it
+    this.extraLabels =
+      inputs.getString("extra-labels") === ""
+        ? inputs.getString("extra-tags")
+        : "";
+    this.spdxExpression = inputs.getString("spdx-expression");
+    this.errorOnConflict = inputs.getBool("error-on-conflict");
+    this.includeOutputPaths = inputs.getBool("include-output-paths");
+    this.rolling = inputs.getBool("rolling");
+    this.mirror = inputs.getBool("mirror");
+    this.name = inputs.getStringOrNull("name");
+    this.rollingMinor = inputs.getNumberOrNull("rolling-minor");
+
+    // Other inputs
+    this.sourceBinary = this.determineSourceBinary;
+  }
+
+  // We first check for a value using the `source-binary` input; if
+  // now-deprecated `flakehub-push-binary`
+  private get determineSourceBinary(): string | null {
+    const sourceBinaryInput = inputs.getStringOrNull("source-binary");
+    const flakeHubPushBinaryInput = inputs.getStringOrNull(
+      "flakehub-push-binary",
+    );
+
+    return sourceBinaryInput !== ""
+      ? sourceBinaryInput
+      : flakeHubPushBinaryInput;
+  }
+
+  private determineVisibility(): Visibility {
+    const visibility = inputs.getString("visibility");
+    if (!VISIBILITY_OPTIONS.includes(visibility)) {
+      actionsCore.setFailed(
+        `Visibility option \`${visibility}\` not recognized. Available options: ${VISIBILITY_OPTIONS.map((opt) => `\`${opt}\``).join(", ")}.`,
+      );
+    }
+    return visibility as Visibility;
+  }
+
+  private async executionEnvironment(): Promise<ExecutionEnvironment> {
+    const env: ExecutionEnvironment = {};
+
+    env.FLAKEHUB_PUSH_VISIBILITY = this.visibility;
+    env.FLAKEHUB_PUSH_TAG = this.tag;
+    env.FLAKEHUB_PUSH_HOST = this.host;
+    env.FLAKEHUB_PUSH_LOG_DIRECTIVES = this.logDirectives;
+    env.FLAKEHUB_PUSH_LOGGER = this.logger;
+    env.FLAKEHUB_PUSH_GITHUB_TOKEN = this.gitHubToken;
+    env.FLAKEHUB_PUSH_REPOSITORY = this.repository;
+    env.FLAKEHUB_PUSH_DIRECTORY = this.directory;
+    env.FLAKEHUB_PUSH_GIT_ROOT = this.gitRoot;
+    // not included: the now-deprecated FLAKEHUB_PUSH_EXTRA_TAGS
+    env.FLAKEHUB_PUSH_EXTRA_LABELS = this.extraLabels;
+    env.FLAKEHUB_PUSH_SPDX_EXPRESSION = this.spdxExpression;
+    env.FLAKEHUB_PUSH_ERROR_ON_CONFLICT = this.errorOnConflict.toString();
+    env.FLAKEHUB_PUSH_INCLUDE_OUTPUT_PATHS = this.includeOutputPaths.toString();
+    env.FLAKEHUB_PUSH_ROLLING = this.rolling.toString();
+    env.FLAKEHUB_PUSH_MIRROR = this.mirror.toString();
+
```

Comment:
is (null).toString() "safe"?

Reply:
In this case they're both Booleans with default values, so this is safe.